src/metrics/chaoss.ts

```diff
@@ -295,3 +295,66 @@ FORMAT JSONCompact`;
     }
   });
 };
+
+interface IssueResolutionDurationOptions {
+  by: 'open' | 'close';
+  type: 'avg' | 'median';
+  unit: 'week' | 'day' | 'hour' | 'minute';
+}
+export const chaossIssueResolutionDuration = async (config: QueryConfig<IssueResolutionDurationOptions>) => {
+  config = getMergedConfig(config);
+  const whereClauses: string[] = ["type = 'IssuesEvent'"];
+  const repoWhereClause = getRepoWhereClauseForClickhouse(config);
+  if (repoWhereClause) whereClauses.push(repoWhereClause);
+  
```

Comment:
Do not add time range condition here, since we need to search the whole database for issue open and close time. If users config `by` to `open`, then `opened_at` should be in the config time range, but not necessary for `closed_at`, so the condition should be after `WHERE` clause and in `HAVING` clause to filter the result.

Reply:
I don't quite get the meaning of the filter `by open_at` or `by close_at`, shouldn't the **issue resolution duration** be the avg/median duration timeframe from when issues are **opened** to when issues are **closed**?

What does it mean if I use the filter `close`?

Reply:
We need to aggregate issues into different timeframe. Like if you want to get the issue resolution duration in 2021, so you need to point out whether you mean all the issues opened in 2021 or closed in 2021.

The parameter is also mentioned in the original doc: https://chaoss.community/metric-issue-resolution-duration/